contracts/near/src/state.rs

```diff
@@ -0,0 +1,155 @@
+    /// The current aggregation price.
+    price:                  i64,
+    /// Confidence interval around the current aggregation price.
+    conf:                   u64,
+    /// Exponentially moving average price.
+    ema_price:              i64,
+    /// Exponentially moving average confidence interval.
+    ema_conf:               u64,
+    /// Price of previous aggregate with Trading status.
+    prev_price:             i64,
+    /// Confidence interval of previous aggregate with Trading status.
+    prev_conf:              u64,
+    /// Publish time of previous aggregate with Trading status.
+    prev_publish_time:      u64,
+}
+
+impl From<&PriceAttestation> for PriceFeed {
+    fn from(price_attestation: &PriceAttestation) -> Self {
+        Self {
+            id:                 price_attestation.price_id,
+            status:             price_attestation.status,
+            publish_time:       price_attestation.publish_time,
+            expo:               price_attestation.expo,
+            max_num_publishers: price_attestation.max_num_publishers,
+            num_publishers:     price_attestation.num_publishers,
+            product_id:         price_attestation.product_id,
+            price:              price_attestation.price,
+            conf:               price_attestation.conf,
+            ema_price:          price_attestation.ema_price,
+            ema_conf:           price_attestation.ema_conf,
+            prev_price:         price_attestation.prev_price,
+            prev_conf:          price_attestation.prev_conf,
+            prev_publish_time:  price_attestation.prev_publish_time.try_into().unwrap(),
+        }
+    }
+}
+
+/// A wrapper around a 16bit chain identifier. We can't use Chain from the Wormhole SDK as it does
+/// not provide borsh serialization but we can re-wrap it here relying on the validation from
+/// `wormhole::Chain`.
+#[derive(
+    BorshDeserialize,
+    BorshSerialize,
+    Clone,
+    Default,
+    Deserialize,
+    Eq,
+    Hash,
+    PartialEq,
+    PartialOrd,
+    Serialize,
+)]
+#[serde(crate = "near_sdk::serde")]
+pub struct Chain(pub u16);
+
+impl From<WormholeChain> for Chain {
+    fn from(chain: WormholeChain) -> Self {
+        Self(chain as u16)
+    }
+}
+
+/// A `Source` describes an origin chain from which Pyth attestations are allowed.
+///
+/// This allows for example Pyth prices to be sent from either Pythnet or Solana, but can be used
+/// to add any additional trusted source chains.
+#[derive(
+    BorshDeserialize,
+    BorshSerialize,
+    Clone,
+    Default,
+    Deserialize,
+    Eq,
+    Hash,
+    PartialEq,
+    PartialOrd,
+    Serialize,
+)]
+#[serde(crate = "near_sdk::serde")]
+pub struct Source {
+    pub emitter:            [u8; 32],
+    pub pyth_emitter_chain: Chain,
+}
+
+/// A local `Vaa` type converted to from the Wormhole definition, this helps catch any upstream
+/// changes to the Wormhole VAA format.
+pub struct Vaa<P> {
```

Comment:
It's clever to have it here! On eth we don't have it and I think the assumption is that they will never ever break it unless it becomes a new version with other apis.